internal/agent/harness/agentic.go

```
@@ -0,0 +1,301 @@
+			}
+		}
+		verdict.HardViolations = sortedKeys(hvSet)
+
+		action, shouldContinue, _ := RouteSufficiencyVerdict(verdict, mode.Label, cycle, mode.MaxOrchestratorCycles, boost)
+
+		// Stagnation guard: override CONTINUE with a partial answer when the
+		// score has not meaningfully improved.
+		if shouldContinue && (verdict.Status == "INSUFFICIENT" || verdict.Status == "USEFUL_BUT_INCOMPLETE") {
+			if prevScore >= 0 && cycle >= stagnationCycles && verdict.Score-prevScore < stagnationGain {
+				action = "ANSWER_PARTIAL"
+				shouldContinue = false
```

golangci-lint (2.12.2)

[error] ineffectual assignment to shouldContinue

(ineffassign)